Dashboard redesign

BenjaminGrabow opened the pull request:
##  What does this PR do
This Pull Request is for redesign the dashboard / refactor the code of the charts / adding 2 graphs and fixing the modul bug in the calendar

##  Description of task to be completed

- redesign the dashboard with Ant Design system

- refactor the code of the charts

- add 2 graphs from the ApexCharts library

- fix the modul bug

##  What are the relevant Trello board stories
https://trello.com/c/u4q5keAv/219-exercise-modal-is-recreated-multiple-times-when-you-click-on-the-exercise-information-for-a-specific-date-in-the-calendar

https://trello.com/c/JCcE2qKX/216-new-design-for-dashboard-with-ant-design

##  How should this manually be tested

- start local server

- go to the react app

- terminal: git checkout dashboard-redesign

- npm install

- npm start
![screencapture-localhost-3000-dashboard-2019-09-12-23_33_45](https://user-images.githubusercontent.com/45399252/64823313-657e0580-d5b7-11e9-9177-2ffd6304d561.png)

sampler36 commented:
Good code and design

shaunmcarmody commented:
Great PR @BenjaminGrabow this is a 100% improvement on what we already have, although a few points where made in the above post I think this should be merged regardless because no breaking changes are present, if we feel like we can iterate on this component and optimise the UX further we will do an A/B test with David Monson.

BenjaminGrabow commented:
> Hey Ben this looks great!
> Wanted to point out a few things based on the screen shot.
> 
> 1. I noticed the monthly pie chart is different in design (has a hole) than than the left and right charts.
> 2. Also, the middle chart is getting a background color of red but that color isn't used anywhere else in the view. Is that meant to be a one off?
> 3. Is the border radius for all card components similar? It appears Calendar gets .6rem but couldn't find it on the others.
> 4. The card shadow for the calendar card and history card appear more pronounced than the pie charts and line charts. Is that intended?
> 5. I have a 13 inch screen, and the calendar component has a hard time fitting in (see attached image). Also, there is an extra week in October that's showing up.
>    ![Snip20190913_1](https://user-images.githubusercontent.com/5153613/64847697-a3fbda80-d620-11e9-9fcd-39c73da21935.png)
> 6. For the workout history section, my personal recommendation would be to make it a list component instead of cards. As it is right now, it could be confusing to read (left to right or top to bottom). Especially if user is on tablet and the cards shrink to 2x2 instead of 3x2.
>    ![Snip20190913_2](https://user-images.githubusercontent.com/5153613/64849132-c3483700-d623-11e9-9272-891505bf5a66.png)

Making right now a new Pull Request with these changes (for desktop to only 750px):

1. Changed the Doghnut also to Pie chart.

2. I added these colours because white looked even more weird. I am not that great in designing and not sure about our colour palette - do you have any suggestions for these background colors ? 

3. Changed the border radius for the charts.

4. Changes shadow for the charts.

5. Some months have 6 weeks (starting at week one and ending in week 6) . This is the layout for the Ant Design calendar - Think its not changeable and would also look weird if the user clicks next monhts and a bigger or smalle height appears of the calendar component. 
The calendar is really big you are right - hope its now better - if not I can change the height of the dates. I also think the heat map would be good to implement for a screen width under 800 or 750px because of the height of the calendar.

6. Added a media query to the History component - hope its be better now.

I also deleted the line charts, Maybe we could add them if I make also the third one for Weekly like for a slider where the user can click on a arrow and he sees instead of the Pie charts the line charts. We could discuss this in a team meeting - would be of course also fine if we dont add them to the project. I made them yesterday evening only for fun because I never did a Line chart and was curious to see how it works and I had the intend to implement the line chart for our User weight chart but heard now that we dont implement the feature.
